src/overlays/actors/ovl_En_Zo/z_en_zo.c

```diff
@@ -37,59 +310,488 @@ const ActorInit En_Zo_InitVars = {
+                return 0x4011;
+            }
+            break;
+    }
+    return 0x4006;
+}
+
+s16 func_80B61298(GlobalContext* globalCtx, Actor* thisx) {
+    switch (func_8010BDBC(&globalCtx->msgCtx)) {
+        case 0:
+        case 1:
+        case 3:
+        case 6:
+        case 7:
+        case 8:
+        case 9:
+            return 1;
+
+        case 2:
+            switch (thisx->textId) {
+                case 0x4020:
+                case 0x4021:
+                    return 0;
+                case 0x4008:
+                    gSaveContext.infTable[18] |= 0x10;
+                    break;
+                case 0x402F:
+                    gSaveContext.infTable[18] |= 0x200;
+                    break;
+            }
+            gSaveContext.eventChkInf[3] |= 1;
+            return 0;
+
+        case 4:
+            switch (func_80106BC8(globalCtx)) {
+                case 0:
+                    return 1;
+                default:
+                    if (thisx->textId == 0x400C) {
+                        thisx->textId = (globalCtx->msgCtx.choiceIndex == 0) ? 0x400D : 0x400E;
+                        func_8010B720(globalCtx, thisx->textId);
+                    }
+                    break;
+            }
+            return 1;
+
+        case 5:
+            switch (func_80106BC8(globalCtx)) {
+                case 0:
+                    return 1;
+                default:
+                    return 2;
+            }
+    }
+
+    return 1;
+}
+
+void EnZo_Blink(EnZo* this) {
+    if (DECR(this->blinkTimer) == 0) {
+        this->eyeTexture++;
+        if (this->eyeTexture >= 3) {
+            this->blinkTimer = Math_Rand_S16Offset(30, 30);
+            this->eyeTexture = 0;
+        }
+    }
+}
+
+void EnZo_Dialog(EnZo* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    this->unk_194.unk_18 = player->actor.posRot.pos;
+    if (this->actionFunc == EnZo_Standing) {
+        // Look down at link if young, look up if old
+        this->unk_194.unk_14 = LINK_IS_CHILD ? 10.0f : -10.0f;
+    } else {
+        this->unk_194.unk_18.y = this->actor.posRot.pos.y;
+    }
+    func_80034A14(&this->actor, &this->unk_194, 11, this->unk_64C);
+    if (this->canSpeak == true) {
+        func_800343CC(globalCtx, &this->actor, &this->unk_194.unk_00, this->dialogRadius, func_80B61024, func_80B61298);
+    }
+}
+
+s32 EnZo_PlayerInProximity(EnZo* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    Vec3f surfacePos;
+    f32 yDist;
+    f32 hDist;
+
+    surfacePos.x = this->actor.posRot.pos.x;
+    surfacePos.y = this->actor.posRot.pos.y + this->actor.waterY;
+    surfacePos.z = this->actor.posRot.pos.z;
+
+    hDist = Math_Vec3f_DistXZ(&surfacePos, &player->actor.posRot.pos);
+    yDist = fabsf(player->actor.posRot.pos.y - surfacePos.y);
+
+    if (hDist < 240.0f && yDist < 80.0f) {
+        return 1;
+    }
+    return 0;
+}
+
+void EnZo_SetAnimation(EnZo* this) {
+    s32 animID = 8;
+
```

Comment:
tiny thing but we do `Id` everywhere else to follow the camel case style. I kinda like ID more tbh but just to keep it consistent